packages/mdc-textfield/README.md

```diff
@@ -329,102 +329,155 @@ Mixin | Description
 --- | ---
 `mdc-text-field-fullwidth-bottom-line-color($color)` | Customizes the fullwidth text field variant bottom line color.
 
-## `MDCTextField` Properties and Methods
+<!-- docgen-tsdoc-replacer:start __DO NOT EDIT, This section is automatically generated__ -->
+### MDCTextField
+#### Methods
 
-Property | Value Type | Description
---- | --- | ---
-`value` | `string` | Proxies to the foundation's `getValue`/`setValue` methods.
-`disabled` | `boolean` | Proxies to the foundation's `isDisabled`/`setDisabled` methods.
-`useNativeValidation` | `boolean` (write-only) | Proxies to the foundation's `setUseNativeValidation` method.
-`valid` | `boolean` | Proxies to the foundation's `isValid`/`setValid` methods.
-`helperTextContent` | `string` (write-only)| Proxies to the foundation's `setHelperTextContent` method when set.
-`ripple` | `MDCRipple` (write-only) | The `MDCRipple` instance for the root element that `MDCTextField` initializes; this only applies to the default Text Field, and is `null` for other variants.
-`leadingIconAriaLabel` | `string` (write-only) | Proxies to the foundation's `setLeadingIconAriaLabel` method.
-`trailingIconAriaLabel` | `string` (write-only) | Proxies to the foundation's `setTrailingIconAriaLabel` method.
-`leadingIconContent` | `string` (write-only) | Proxies to the foundation's `setLeadingIconContent` method.
-`trailingIconContent` | `string` (write-only) | Proxies to the foundation's `setTrailingIconContent` method.
-
-In addition to the above, the following properties proxy to the `input` element's properties of the same name:
-
-* `required`
-* `pattern`
-* `minLength`
-* `maxLength`
-* `min`
-* `max`
-* `step`
-
-Method Signature | Description
+Signature | Description
 --- | ---
+`emit(evtType: string, evtData: T, shouldBubble?: boolean) => void` | Fires a cross-browser-compatible custom event from the component root of the given type, with the given data.
```

Comment:
We might've to ignore emit, listen and unlisten which are not useful to clients.

Reply:
Technically a user can use the emit, listen, and unlisten methods. I think leaving this out would make the list incomplete.